lib/ProductOpener/Ingredients.pm

```diff
@@ -1277,6 +1278,231 @@ sub match_origin_of_the_ingredient_origin ($ingredients_lc, $text_ref, $matched_
 	return 0;
 }
 
+=head2 parse_processing_from_ingredient ( $ingredients_processing_regexps, $ingredients_lc, $processing, $ingredient, $ingredient_id )
+
+This function extract processing method from one ingredient.
+If processing methods are found and remaining ingredient text exists without the processing method,
+then, it returns:
+	- $processing (concatenate if more than one), 
+	- $ingredient (without processing) and 
+	- $ingredient_id (without processing)
+If it does not result in known ingredient, then it returns the same but unchanged.
+
+=head3 Arguments
+
+=head4 ingredients_processing_regexps
+
+list of regexps with each synonyms of all ingredients processes
+
+=head4 ingredients_lc
+
+language abbreviation (en for English, for example)
+
+=head4 processing
+
+", " concatenated processing methods. string
+
+=head4 ingredient
+
+string ("pear", for example)
+
+=head4 ingredient_id
+
+English first element for that ingredient (en:pear, for example)
+
+=head4 and
+
+hashmap of regex for "and" per languages
+
+=head4 ingredient_recognized
+
+0 or 1
+
+=head3 Return values
+
+=head4 processing
+
+space concatenated processing methods
+
+=head4 ingredient
+
+updated ingredient without processing methods
+
+=head4 ingredient_id
+
+English first element for that ingredient (en:pear, for example)
+
+=head4 ingredient_recognized
+
+0 or 1
+
+=cut
+
+sub parse_processing_from_ingredient ($ingredients_processing_regexps,
```

Comment:
It's a bit confusing to have so many input parameters. When the function is called, it's difficult to remember which parameter is what.

I think we can keep only $ingredients_lc and $ingredient. It forces us to canonicalize the ingredient at the beginning of the function, but it simplifies things a lot I think.

I made a PR here with a proposed refactor: https://github.com/openfoodfacts/openfoodfacts-server/pull/9146